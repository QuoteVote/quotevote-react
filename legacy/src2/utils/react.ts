import React, { 
  useState, 
  useEffect, 
  useRef, 
  ReactNode, 
  ChangeEvent, 
  KeyboardEvent,
  FormEvent,
  MouseEvent,
  Dispatch,
  SetStateAction
} from 'react';

export type { 
  ReactNode,
  ChangeEvent,
  KeyboardEvent,
  FormEvent,
  MouseEvent,
  Dispatch,
  SetStateAction
};

export { 
  React as default,
  useState,
  useEffect,
  useRef
};

// Add type declarations for common event handlers
export type InputChangeEvent = ChangeEvent<HTMLInputElement>;
export type TextAreaChangeEvent = ChangeEvent<HTMLTextAreaElement>;
export type ButtonClickEvent = MouseEvent<HTMLButtonElement>;
export type SetState<T> = Dispatch<SetStateAction<T>>;

// Add type declarations for common component props
export interface BaseProps {
  className?: string;
  style?: React.CSSProperties;
  children?: ReactNode;
}

export interface InputProps extends BaseProps {
  value: string;
  onChange: (value: string) => void;
  placeholder?: string;
  disabled?: boolean;
} 