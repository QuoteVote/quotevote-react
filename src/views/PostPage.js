/* eslint-disable no-underscore-dangle */
/* eslint-disable prefer-destructuring */
import React, { useState } from 'react'

import Button from 'mui-pro/CustomButtons/Button'
import Card from 'mui-pro/Card/Card'
import CardBody from 'mui-pro/Card/CardBody'
import CardFooter from 'mui-pro/Card/CardFooter'
import CardHeader from 'mui-pro/Card/CardHeader'
import GridContainer from 'mui-pro/Grid/GridContainer'
import GridItem from 'mui-pro/Grid/GridItem'
import Snackbar from 'mui-pro/Snackbar/Snackbar'

import Divider from '@material-ui/core/Divider'
import CardActions from '@material-ui/core/CardActions'

import VotingBoard from 'components/VotingComponents/VotingBoard'
import VotingPopup from 'components/VotingComponents/VotingPopup'
import ApproveRejectPopover from 'hhsbComponents/ApproveRejectPopover'

// import Content from '../components/ContentList';

// import styles from 'assets/jss/material-dashboard-pro-react/views/dashboardStyle';

import FaceIcon from '@material-ui/icons/Face'
import CloseIcon from '@material-ui/icons/Close'
import CheckIcon from '@material-ui/icons/Check'
import { useQuery, useMutation } from '@apollo/react-hooks'
import moment from 'moment'

import PostPageSkeleton from 'views/Skeletons/PostPageSkeleton'
import { useSelector } from 'react-redux'
import { cloneDeep, findIndex } from 'lodash'

import { GET_POST, GET_TOP_POSTS } from 'graphql/query'
import {
  VOTE, ADD_COMMENT, APPROVE_POST, REJECT_POST,
} from 'graphql/mutations'
import Chat from 'assets/img/Chat.svg'
import Heart from 'assets/img/Heart.svg'
import Send from 'assets/img/Send.svg'

const PostPage = () => {
  // const url = window.location.href
  // const urlSegment = url.split('/')
  // const domain = urlSegment[5]
  // const contentId = urlSegment[6]

  const [selectedText, setSelectedText] = useState('')
  const [anchorEl, setAnchorEl] = useState(null)
  const [snackBar, setSnackBar] = useState({ open: false, message: '', color: 'danger' })
  const [buttonType, setButtonType] = useState('approved')
  const user = useSelector((state) => state.user.data)
  const postId = useSelector((state) => state.ui.selectedPost.id)
  const [addVote] = useMutation(VOTE, {
    update(
      cache,
      {
        // eslint-disable-next-line no-shadow
        data: { addVote },
      }
    ) {
      const data = cache.readQuery({
        query: GET_POST,
        variables: { postId },
      })
      const clonedPost = cloneDeep(data)

      const index = findIndex(clonedPost.post.votedBy, (vote) => vote.userId === user._id)
      if (index !== -1) {
        clonedPost.post.votedBy[index].type = addVote.type
        clonedPost.post.upvotes = addVote.type === 'up' ?
          clonedPost.post.upvotes + 1 : clonedPost.post.upvotes - 1

        clonedPost.post.downvotes = addVote.type === 'down' ?
          clonedPost.post.downvotes + 1 : clonedPost.post.downvotes - 1
      } else {
        clonedPost.post.votedBy.push({ type: addVote.type, userId: user._id })
        if (addVote.type === 'up') {
          clonedPost.post.upvotes++
        } else {
          clonedPost.post.downvotes++
        }
      }
      cache.writeQuery({
        query: GET_POST,
        variables: { postId },
        data: { ...clonedPost },
      })
    },
    refetchQueries: [
      {
        query: GET_TOP_POSTS,
        variables: { limit: 5, offset: 0, searchKey: '' },
      },
    ],
  })

  const [addComment] = useMutation(ADD_COMMENT, {
    refetchQueries: [
      {
        query: GET_TOP_POSTS,
        variables: { limit: 5, offset: 0, searchKey: '' },
      },
      {
        query: GET_POST,
        variables: { postId },
      },
    ],
  })

  const [approvePost] = useMutation(APPROVE_POST, {
    refetchQueries: [
      {
        query: GET_POST,
        variables: { postId },
      },
    ],
  })

  const [rejectPost] = useMutation(REJECT_POST, {
    refetchQueries: [
      {
        query: GET_POST,
        variables: { postId },
      },
    ],
  })
  // const classes = useStyles();

  const { loading, error, data } = useQuery(GET_POST, {
    variables: { postId },
  })

  if (loading) return <PostPageSkeleton />

  if (error) return `Something went wrong: ${error}`

  const { post } = data

  const disableApproveReject = user._id === post.userId
  const disablApprove = post.approvedBy.includes(user._id)
  const disableReject = post.rejectedBy.includes(user._id)

  const handleVoting = async (type) => {
    const vote = {
      // text: selectedText.text,
      postId: data.post._id,
      userId: user._id,
      type,
      startWordIndex: selectedText.startIndex,
      endWordIndex: selectedText.endIndex,
    }
    try {
      await addVote({ variables: { vote } })
      setSnackBar({
        open: true,
        message: 'Voted Successfully',
        color: 'success',
      })
    } catch (err) {
      setSnackBar({
        open: true,
        message: `Vote Error: ${err.message}`,
        color: 'danger',
      })
    }
  }

  const handleAddComment = async (comment, commentWithQuote = false) => {
    let startIndex; let endIndex; let
      quoteText

    // const HASHTAGS_REGEX = /#(\w|\d)+/g
    // const hashtags = comment.match(HASHTAGS_REGEX)

    if (selectedText) {
      startIndex = selectedText.startIndex
      endIndex = selectedText.endIndex
      quoteText = selectedText.text
    } else {
      startIndex = 0
      endIndex = 0
      quoteText = ''
    }

    // TODO: ommit quote props if user did not highlight text
    const newComment = {
      // contentId,
      // creatorId: content.creatorId,
      userId: user._id,
      content: comment,
      startWordIndex: startIndex,
      endWordIndex: endIndex,
      postId,
      url: post.url,
      // hashtags,
      quote: commentWithQuote ? quoteText : '',
    }

    try {
      await addComment({ variables: { comment: newComment } })
      setSnackBar({
        open: true,
        message: 'Commented Successfully',
        color: 'success',
      })
    } catch (err) {
      setSnackBar({
        open: true,
        message: `Comment Error: ${err.message}`,
        color: 'danger',
      })
    }
  }

  const handlePopoverOpen = (event, type) => {
    setAnchorEl(event.currentTarget)
    setButtonType(type)
  }

  const handlePopoverClose = () => {
    setAnchorEl(null)
  }

  const handleApprovePost = async () => {
    if (!disableApproveReject || !disablApprove) {
      try {
        setSnackBar({
          open: true,
          message: 'Approved Post Successfully',
          color: 'success',
        })
        approvePost({ variables: { userId: user._id, postId: post._id } })
      } catch (err) {
        setSnackBar({
          open: true,
          message: `Approve Post Error: ${err.message}`,
          color: 'danger',
        })
      }
    }
  }

  const handleRejectPost = async () => {
    if (!disableApproveReject || !disableReject) {
      try {
        rejectPost({ variables: { userId: user._id, postId: post._id } })
        setSnackBar({
          open: true,
          message: 'Rejected Post Successfully',
          color: 'success',
        })
      } catch (err) {
        setSnackBar({
          open: true,
          message: `Reject Post error: ${err.message}`,
          color: 'danger',
        })
      }
    }
  }

  return (
    <div>
      <GridContainer spacing={1} direction="col">
        <GridItem xs={6}>
          <Card style={{ height: '800px' }}>
            <CardHeader style={{ zIndex: 0 }}>
              <div
                style={{
                  display: 'flex',
                  direction: 'row',
                  justifyContent: 'space-between',
                  zIndex: 0,
                }}
              >
                <div
                  style={{
                    display: 'flex',
                    direction: 'row',
                    alignContent: 'center',
                    justifyContent: 'space-between',
                  }}
                >
                  <p
                    style={{
                      color: '#E91E63',
                      fontSize: '25px',
                      font: 'League Spartan',
                      fontWeight: 'bold',
                    }}
                  >
                    {post.title}
                  </p>
                  <img
                    alt=""
                    src={Chat}
                    style={{ height: '20px', paddingLeft: '10px' }}
                  />
                </div>
                <div
                  style={{
                    display: 'flex',
                    direction: 'row',
                    justifyContent: 'flex-end',
                    flexBasis: '100px',
                  }}
                >
                  <p>
                    <strong style={{ color: 'green' }}>+</strong>
                    {post.upvotes}
                    /
                    <strong style={{ color: 'red' }}>-</strong>
                    {post.downvotes}
                  </p>
                  <img
                    alt="Send icon"
                    src={Send}
                    style={{
                      height: '15px',
                      paddingLeft: '15px',
                      paddingTop: '3px',
                    }}
                  />
                  <img
                    alt="Heart icon"
                    src={Heart}
                    style={{
                      height: '15px',
                      paddingLeft: '15px',
                      paddingTop: '3px',
                    }}
                  />
                </div>
              </div>
              <Divider />
            </CardHeader>
            <CardBody>
              <VotingBoard
                content={post.text}
                onSelect={setSelectedText}
                selectedText={selectedText}
              >
                {({ text }) => (
                  <VotingPopup
                    onVote={handleVoting}
                    onAddComment={handleAddComment}
                    text={text}
                    selectedText={selectedText}
                    votedBy={post.votedBy}
                  />
                )}
              </VotingBoard>
            </CardBody>
            <CardActions>
              <GridContainer spacing={4} justify="center">
                <GridItem>
                  <Button
                    variant="contained"
                    startIcon={<CloseIcon />}
                    onMouseEnter={(e) => handlePopoverOpen(e, 'rejected')}
                    onClick={handleRejectPost}
                    style={{
                      backgroundColor: '#f44336',
                      color: 'white',
                      opacity: disableApproveReject || disableReject ? 0.65 : 1,
                    }}
                  >
                    { disableReject ? 'REJECTED' : 'REJECT' }
                  </Button>
                </GridItem>
                <GridItem>
                  <Button
                    variant="contained"
                    startIcon={<CheckIcon />}
                    onMouseEnter={(e) => handlePopoverOpen(e, 'approved')}
                    onClick={handleApprovePost}
                    style={{
                      backgroundColor: '#4caf50',
                      color: 'white',
                      opacity: disableApproveReject || disablApprove ? 0.65 : 1,
                    }}
                  >
                    { disablApprove ? 'APPROVED' : 'APPROVE' }
                  </Button>
                </GridItem>
              </GridContainer>
            </CardActions>
          </Card>
        </GridItem>
        <GridItem xs={6} style={{ paddingBottom: 0 }}>
          <Card>
            <CardHeader>
              <div
                style={{
                  display: 'flex',
                  direction: 'row',
                  justifyContent: 'space-between',
                }}
              >
                <p
                  style={{
                    color: '#E91E63',
                    fontSize: '25px',
                    font: 'League Spartan',
                    fontWeight: 'bold',
                  }}
                >
                  Comments
                </p>
              </div>
              <Divider />
            </CardHeader>
          </Card>
          {post.comments
            .sort((a, b) => moment(b.created).diff(moment(a.created)))
            .map((comment, index) => (
              <Card key={`comment-${index}`}>
                <CardBody>
                  <p style={{ fontSize: 12 }}>
                    <span
                      style={{
                        height: 60,
                        width: 60,
                        backgroundColor: '#df2769',
                        float: 'left',
                        margin: '0px 10px 10px 0px',
                        textAlign: 'center',
                        borderRadius: 3,
                        paddingTop: 17,
                      }}
                    >
                      <FaceIcon />
                    </span>
                    <h5 style={{ margin: 0 }}>username</h5>
                    {comment.content}
                  </p>
                </CardBody>
                <CardFooter chart testimonial>
                  <span style={{ float: 'right' }}>
                    {moment(comment.created).format('MM/DD/YYYY hh:mm A')}
                  </span>
                </CardFooter>
              </Card>
            ))}
        </GridItem>
      </GridContainer>
      <ApproveRejectPopover
        anchorEl={anchorEl}
        handlePopoverClose={handlePopoverClose}
        type={buttonType}
        approvedBy={post.approvedBy}
        rejectedBy={post.rejectedBy}
      />
      <Snackbar
        place="bc"
        color={snackBar.color}
        message={snackBar.message}
        open={snackBar.open}
        closeNotification={() => setSnackBar({ open: false, message: '' })}
        close
      />
    </div>
  )
}

export default PostPage
