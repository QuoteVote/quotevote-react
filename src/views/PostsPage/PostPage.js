import React, { useEffect, useState } from 'react'
import PropTypes from 'prop-types'
import { Grid, Typography } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'
import { useQuery, useSubscription } from '@apollo/react-hooks'
import { useSelector } from 'react-redux'
import { isEmpty } from 'lodash'
import Link from '@material-ui/core/Link'
import Post from '../../components/Post/Post'
import PostActionList from '../../components/PostActions/PostActionList'
import PostSkeleton from '../../components/Post/PostSkeleton'
import { GET_ROOM_MESSAGES, GET_POST } from '../../graphql/query'
import { NEW_MESSAGE_SUBSCRIPTION } from '../../graphql/subscription'
import PostChatSend from '../../components/PostChat/PostChatSend'

const useStyles = makeStyles(() => ({
  root: {
    marginTop: 10,
  },
  link: {
    color: '#00bcd4',
  },
}))

function PostPage({ postId }) {
  const classes = useStyles()
  const [postHeight, setPostHeight] = useState()

  const idSelector = useSelector((state) => state.ui.selectedPost.id)
  const user = useSelector((state) => state.user.data)

  const {
    loading: loadingPost,
    error: postError,
    data: postData,
    refetch: refetchPost,
  } = useQuery(GET_POST, {
    variables: { postId },
  })

  const { post } = !loadingPost && postData

  // To reset the scroll when the selected post changes
  useEffect(() => {
    window.scrollTo(0, 0)
    if (post) {
      setPostHeight(document.getElementById('post').clientHeight)
    }
  }, [post])

  useEffect(() => {
    refetchPost({ postId: idSelector })
  }, [idSelector, refetchPost])

  let messageRoomId
  let title
  if (post) {
    messageRoomId = postData.post.messageRoom._id
    title = postData.post.title
  }

  const {
    loading: loadingMessages, data: messageData, refetch: refetchMessages,
  } = useQuery(GET_ROOM_MESSAGES, {
    skip: !messageRoomId,
    variables: { messageRoomId },
  })

  useSubscription(
    NEW_MESSAGE_SUBSCRIPTION,
    {
      skip: !messageRoomId,
      variables: { messageRoomId },
      onSubscriptionData: async () => {
        await refetchMessages()
      },
    },
  )

  if (postError) return 'Something went wrong!'

  const { messages } = (!loadingMessages && messageData) || []

  const {
    comments, votes, quotes, postUrl,
  } = post || {
    comments: [], votes: [], quotes: [], postUrl: '',
  }
  let postActions = []

  const { url } = !loadingPost && postUrl

  if (!isEmpty(comments)) {
    postActions = postActions.concat(comments)
  }

  if (!isEmpty(votes)) {
    postActions = postActions.concat(votes)
  }

  if (!isEmpty(quotes)) {
    postActions = postActions.concat(quotes)
  }

  if (!isEmpty(messages)) {
    postActions = postActions.concat(messages)
  }

  if (!loadingPost && !post) {
    return (
      <Grid
        container
        direction="row"
        justify="space-around"
        alignItems="flex-start"
        spacing={4}
        className={classes.root}
        style={{ position: 'relative' }}
      >
        <Grid item xs={12}>
          <Typography>
            Invalid post.
            {' '}
            <Link href="/home" className={classes.link}>
              Return to homepage.
            </Link>
          </Typography>
        </Grid>
      </Grid>
    )
  }

  return (
    <Grid
      container
      direction="row"
      justify="space-around"
      alignItems="flex-start"
      spacing={4}
      className={classes.root}
      style={{ position: 'relative' }}
    >
      <Grid item xs={12} md={6} id="post">
        {loadingPost ? <PostSkeleton /> : <Post post={post} loading={loadingPost} user={user} postHeight={postHeight} postActions={postActions} />}
      </Grid>
      <Grid item className={classes.root} xs={12} md={6}>
        <PostChatSend messageRoomId={messageRoomId} title={title} />
        <PostActionList loading={loadingPost} postActions={postActions} postUrl={url} />
      </Grid>
    </Grid>
  )
}

PostPage.propTypes = {
  postId: PropTypes.string,
}

export default PostPage
