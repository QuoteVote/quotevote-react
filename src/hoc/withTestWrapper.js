import React from 'react'
import { PersistGate } from 'redux-persist/integration/react'
import { ApolloProvider } from '@apollo/react-hooks'
import { Provider } from 'react-redux'
import { createTheme, ThemeProvider } from '@material-ui/core/styles'
import { Router } from 'react-router-dom'
import { createBrowserHistory } from 'history'
import store, { persistor } from '../store/store'
import client from '../config/apollo'
import customTheme from '../theme'
const hist = createBrowserHistory()
const theme = createTheme(customTheme)

const withTestWrapper = (Component) => (props) => (
  <ApolloProvider client={client}>
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <ThemeProvider theme={theme}>
          <Router history={hist}>
            <Component {...props} />
          </Router>
        </ThemeProvider>
      </PersistGate>
    </Provider>
  </ApolloProvider>
)
export default withTestWrapper
