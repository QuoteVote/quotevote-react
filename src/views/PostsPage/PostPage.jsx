import { useEffect, useState } from 'react'
import PropTypes from 'prop-types'
import { Grid, Typography } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'
import { useQuery, useSubscription } from '@apollo/react-hooks'
import { useSelector } from 'react-redux'
import { isEmpty } from 'lodash'
import { Redirect } from 'react-router-dom'
import Post from '../../components/Post/Post'
import PostActionList from '../../components/PostActions/PostActionList'
import PostSkeleton from '../../components/Post/PostSkeleton'
import { GET_ROOM_MESSAGES, GET_POST } from '../../graphql/query'
import { NEW_MESSAGE_SUBSCRIPTION } from '../../graphql/subscription'
import PostChatSend from '../../components/PostChat/PostChatSend'

const useStyles = makeStyles(() => ({
  root: {
    marginTop: 10,
    minHeight: '100vh',
    boxSizing: 'border-box',
  },
  emptyPost: {
    marginTop: 100,
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
  link: {
    color: '#00bcd4',
  },
}))

function PostPage({ postId }) {
  const classes = useStyles()
  const [postHeight, setPostHeight] = useState()

  const idSelector = useSelector((state) => state.ui.selectedPost.id)
  const user = useSelector((state) => state.user.data)

  const {
    loading: loadingPost,
    error: postError,
    data: postData,
    refetch: refetchPost,
  } = useQuery(GET_POST, {
    variables: { postId },
  })

  const { post } = (!loadingPost && postData) || {}

  // To reset the scroll when the selected post changes
  useEffect(() => {
    window.scrollTo(0, 0)
    if (post) {
      setPostHeight(document.getElementById('post').clientHeight)
    }
  }, [post])

  useEffect(() => {
    refetchPost({ postId: idSelector })
  }, [idSelector, refetchPost])

  let messageRoomId
  let title
  if (post) {
    messageRoomId = postData.post.messageRoom._id
    title = postData.post.title
  }

  const {
    loading: loadingMessages, data: messageData, refetch: refetchMessages,
  } = useQuery(GET_ROOM_MESSAGES, {
    skip: !messageRoomId,
    variables: { messageRoomId },
  })

  useSubscription(
    NEW_MESSAGE_SUBSCRIPTION,
    {
      skip: !messageRoomId,
      variables: { messageRoomId },
      onSubscriptionData: async () => {
        await refetchMessages()
      },
    },
  )

  if (postError) return <Redirect to="/error" />

  const { messages } = (!loadingMessages && messageData) || []

  const {
    comments, votes, quotes,
  } = post || { comments: [], votes: [], quotes: [] }
  let postActions = []

  if (!isEmpty(comments)) {
    postActions = postActions.concat(comments)
  }

  if (!isEmpty(votes)) {
    postActions = postActions.concat(votes)
  }

  if (!isEmpty(quotes)) {
    postActions = postActions.concat(quotes)
  }

  if (!isEmpty(messages)) {
    postActions = postActions.concat(messages)
  }

  if (!loadingPost && !post) {
    return <Redirect to="/error" />
  }

  const { url } = (!loadingPost && post) || {}

  return (
    <Grid
      container
      direction="row"
      justify="space-around"
      alignItems="flex-start"
      spacing={4}
      className={classes.root}
      style={{ position: 'relative' }}
    >
      <Grid item xs={12} md={6} id="post">
        {loadingPost ? <PostSkeleton /> : <Post post={post} loading={loadingPost} user={user} postHeight={postHeight} postActions={postActions} />}
      </Grid>
      <Grid item className={classes.root} xs={12} md={6}>
        <PostChatSend messageRoomId={messageRoomId} title={title} />
        <PostActionList loading={loadingPost} postActions={postActions} postUrl={url} />
      </Grid>
    </Grid>
  )
}

PostPage.propTypes = {
  postId: PropTypes.string,
}

export default PostPage
