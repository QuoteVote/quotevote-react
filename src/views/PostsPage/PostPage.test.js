import React from 'react'
import { render } from '@testing-library/react'

// Component being tested
import PostPage from './PostPage'
import withTestWrapper from '../../hoc/withTestWrapper'

const PostPageWrapper = withTestWrapper(PostPage)

describe('Post Page test -', () => {
  it('renders correctly', () => {
    const { container } = render(
      <PostPageWrapper />,
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
