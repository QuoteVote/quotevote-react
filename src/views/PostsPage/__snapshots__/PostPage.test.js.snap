// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Page test - renders correctly 1`] = `
<div
  class="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-align-items-xs-flex-start MuiGrid-justify-xs-space-around"
  style="position: relative;"
>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
  >
    <div
      class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiCardHeader-root"
      >
        <div
          class="MuiCardHeader-content"
        >
          <span
            class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
            >
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
                  style="width: 200px; margin-right: 10px;"
                />
              </div>
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-circle MuiSkeleton-pulse"
                  style="width: 30px; height: 30px;"
                />
              </div>
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-circle MuiSkeleton-pulse"
                  style="width: 30px; height: 30px;"
                />
              </div>
            </div>
          </span>
        </div>
        <div
          class="MuiCardHeader-action"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
            style="margin-top: 15px;"
          >
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
                style="width: 30px; margin-right: 5px;"
              />
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
               
              /
               
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
                style="width: 30px; margin-left: 5px;"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiCardHeader-root"
        style="padding: 0px; margin-left: 15px;"
      >
        <div
          class="MuiCardHeader-avatar"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-circle MuiSkeleton-pulse"
            style="width: 50px; height: 50px;"
          />
        </div>
        <div
          class="MuiCardHeader-content"
        >
          <span
            class="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
              style="width: 80px; margin-bottom: 5px;"
            />
          </span>
          <span
            class="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
              style="width: 130px;"
            />
          </span>
        </div>
      </div>
      <div
        class="MuiCardContent-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
          style="height: 30px;"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
          style="height: 30px;"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
          style="height: 30px;"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
          style="width: 50%; height: 30px;"
        />
      </div>
      <div
        class="MuiCardActions-root"
        style="margin-left: 5px;"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
          style="width: 130px; height: 30px; margin-right: 5px;"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
          style="width: 130px; height: 30px;"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-circle MuiSkeleton-pulse"
          style="width: 30px; height: 30px; margin-left: auto;"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-circle MuiSkeleton-pulse"
          style="width: 30px; height: 30px; margin-right: 5px; margin-left: 5px;"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-circle MuiSkeleton-pulse"
          style="width: 30px; height: 30px;"
        />
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root makeStyles-root-1 MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
  >
    <div
      class="MuiGrid-root MuiGrid-container"
    >
      <div
        class="MuiGrid-root makeStyles-root-2 MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
        style="width: 100%;"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-md-2"
        >
          <p
            class="MuiTypography-root makeStyles-chat-3 MuiTypography-body1"
          >
            Chat
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-md-10"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              class="MuiInputBase-root makeStyles-input-4"
            >
              <input
                class="MuiInputBase-input"
                placeholder="type a message..."
                type="text"
                value=""
              />
            </div>
            <button
              class="MuiButtonBase-root MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <img
                  alt="send"
                  src="SendIcon.svg"
                />
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-flex-start MuiGrid-justify-xs-space-between"
    >
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root"
          style="margin-left: auto;"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              height="32"
              viewBox="0 0 32 32"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M31.8801 0.478938C31.7142 0.183188 31.4016 0 31.0625 0H0.937492C0.598367 0 0.285617 0.183188 0.119742 0.478938C-0.0460076 0.774688 -0.0393201 1.137 0.137555 1.42631L11.0457 19.2763V28.05C11.0457 28.4408 11.2882 28.7906 11.654 28.9278L19.6873 31.9403C19.794 31.9804 19.9056 32 20.0164 32C20.2048 32 20.3914 31.9433 20.5505 31.8331C20.8033 31.6579 20.9541 31.37 20.9541 31.0625V19.2763L31.8624 1.42638C32.0393 1.137 32.046 0.77475 31.8801 0.478938Z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <span
      class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
      style="height: 118px;"
    />
    <br />
    <span
      class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
      style="height: 118px;"
    />
    <br />
    <span
      class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
      style="height: 118px;"
    />
  </div>
</div>
`;
